src/qtxdg/xdgmimeapps.cpp

```diff
@@ -96,6 +96,7 @@ QList<XdgDesktopFile *> XdgMimeApps::categoryApps(const QString &category)
         if (!categories.isEmpty() && (categories.contains(cat) || categories.contains(QL1S("X-") + cat)))
             dl.append(df);
     }
+    qDeleteAll(apps);
```

Comment:
Doesn't this make `df` consist of dangling pointers? I think we may need to delete only those pointers that aren't included in `df`.

Reply:
You were faster. That was a debug branch to find the source of the leak. Pushed the wrong branch. Fixed now.